### DISCUSSION TITLE
Is there a way to clear the cache?

### DISCUSSION THREAD

**Discussion by regirock365** [urql-graphql/urql#1528] [Q&A] [answered]
Is there a way to clear the cache? There are some mutations in my application that call for a refresh of the cache and I couldn't find such a method in the docs so I thought I'd ask.

**Comment by kitten** [comment, marked as answer]
Specifically which cache? In Graphcache we'd recommend mutations to selectively use `cache.invalidate`, and in the default cache, if you do want to wipe everything, we recommend you to recreate the client, which should automatically rerender everything in your app and refetch all data.

**Reply by regirock365** [reply]
I am using Graphcache. Many queries would essentially need to be reset when a user signs out, for example. I could invalidate the user, but many queries are not directly linked to the user type.

I have approximated this so far by checking if a user is signed in at the top level and showing a login screen if they are not, thus hiding everything else. If the user signs back in, many components are set to re-execute queries when they first render. This is very hacky so I'm wondering if there is a better way to do it, short of resetting the cache.

**Reply by haswalt** [reply]
I handle this by creating a new client based on auth state. So when a user logs in we create a new client and again when they log out. This ensures any caches or state stored in the client are cleared / destroyed preventing any leaks between users.
